api/v1beta1/preflightvalidation_types.go

```diff
@@ -100,6 +71,58 @@ type PreflightValidation struct {
 	Status PreflightValidationStatus `json:"status,omitempty"`
 }
 
+func (p *PreflightValidation) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1beta2.PreflightValidation)
+
+	dst.ObjectMeta = p.ObjectMeta
+	dst.Spec = p.Spec
+
+	dst.Status = v1beta2.PreflightValidationStatus{}
+
+	if count := len(p.Status.CRStatuses); count > 0 {
+		dst.Status.Modules = make([]v1beta2.PreflightValidationModuleStatus, 0, count)
+
+		for k, v := range p.Status.CRStatuses {
+			namespace, name, ok := strings.Cut(k, "/")
+
+			if !ok || v == nil {
+				// Elements whose key is not a namespace name or that are nil are invalid.
+				return nil
```

Comment:
If they are invalid, shouldn't we return an error?

Reply:
The function signature comes from the `Convertible` interface that we need to implement; we cannot change it.
The situation is that none of the current `v1beta1` `PreflightValidation` resources include a namespace. So all of them are technically invalid.
By skipping all invalid elements we can proceed with the conversion. The new `v1beta2` object can then be reconciled with the updated `PreflightValidation` reconciler that handles namespaces. Now statuses will correctly include the namespace.